lib/config/config.go

```diff
@@ -365,17 +409,28 @@ func (c *Config) UpdateFromFile(path string) error {
 // Returns errors encountered when generating or writing the file, or nil
 // otherwise.
 func (c *Config) ToFile(path string) error {
-	var w bytes.Buffer
-	e := toml.NewEncoder(&w)
+	b, err := c.ToBytes()
+	if err != nil {
+		return err
+	}
 
-	t := new(tomlConfig)
-	t.fromConfig(c)
+	return ioutil.WriteFile(path, b, 0644)
+}
 
-	if err := e.Encode(*t); err != nil {
-		return err
+// ToBytes encodes the config into a byte slice. It errors if the encoding
+// fails, which should never happen at all because of generael type safeness.
+func (c *Config) ToBytes() ([]byte, error) {
+	var buffer bytes.Buffer
+	e := toml.NewEncoder(&buffer)
+
+	tc := tomlConfig{}
+	tc.fromConfig(c)
+
+	if err := e.Encode(tc); err != nil {
+		return nil, err
 	}
 
-	return ioutil.WriteFile(path, w.Bytes(), 0644)
+	return buffer.Bytes(), nil
```

Comment:
I think this code is fine, but it would be more conventional to encode to a io.Writer than to require bytes, e.g. `func (c *Config) WriteTo(out io.Writer) error`.

Reply:
Hm, I see your point but for now I would leave it as it is. I thought about removing this function entirely but it might have their use case.